### PR TITLE
Adding recommendations to the aggregate report

### LINKED ISSUE #2199: Adding recommendations
**Description**
Making recommendations visible. 

**How to do this**
- System Specific and Safe Connections both have "ips" in the aggregate report.
- These ips are all the finding_types that belong to those reports. 
- Those finding_types contain a "recommendation", which can be collected in the recommendations chapter of the report.

**Must have**
Collecting recommendations from the finding_types that are currently available in the report (from System Specific and Safe Connections).

**Could have**
Collecting recommendations from the other reports (RPKI and IPv6)

### PR DESCRIPTION
**Pull request by madelondohmen**
### Changes
Recommendations are now collected from the finding_types that where already available in the report (from System Specific and Safe Connections).

Added the following code:
* Collecting recommendations from findings in Safe Connections
* Collecting recommendations from findings in System Specific
* Added recommendations back to the return (was temporarily removed)
* Added a filter to exclude "None" from the collected recommendations
* Updated recommendations.html

### Issue link
Closes #2199 

### Extra info
I've made some changes in the HTML, because total findings (from Vulnerabilities) can be 0, while there are recommendations (from findings in Safe Connections and System Specific). 

### Demo
When there are multiple recommendations collected:
afbeelding

When there are no recommendations:
afbeelding

### Future 
In the future, it would be nice to collect recommendations from the other reports (RPKI and IPv6) as well.

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/report_types/aggregate_organisation_report/recommendations.html [hunk 1]
```diff
@@ -3,9 +3,7 @@
 <section id="recommendations">
     <div>
         <h2>{% translate "Recommendations" %}</h2>
-        {% if total_findings == 0 %}
-            <p>{% translate "There are no recommendations." %}</p>
-        {% else %}
+        {% if total_findings != 0 %}
```
Comment (underdarknl) [inline rocky/reports/report_types/aggregate_organisation_report/recommendations.html:6]: `!= 0` might not be needed as an non empty set/list/ not 0 int evaluates to true

**Review by ammar92 — DISMISSED** [review]
## Checklist for QA:
- [ ] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [ ] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Seems to work as intended

image

**Comment by Donnype** [comment]
@madelondohmen Should we count the recommendations and put this in a table? I found this mock up in the report I'm working on:

Scherm­afbeelding 2023-12-28 om 12 52 36

Ping me if this can easily be added and you want the html :)

**Review by ammar92 — APPROVED** [review]

**Comment by underdarknl** [comment]
> @madelondohmen Should we count the recommendations and put this in a table? I found this mock up in the report I'm working on:
> 
> Scherm­afbeelding 2023-12-28 om 12 52 36
> 
> Ping me if this can easily be added and you want the html :)

That would be lovely, and would also provide a location to (in the future) add the dropdown box with the different locations where each recommendation should be applied.

**Review by underdarknl — APPROVED** [review]

**Pull request merged**